src/components/GlobalState.js

```diff
@@ -41,6 +46,7 @@ export const useGlobalStateContext = () => useContext(GlobalStateContext);
 
 export const GlobalStateProvider = props => {
   const [state, setState] = useState({
+    githubToken: getGithubToken(),
```

Comment:
Not sure about the advantages of tracking the githubtoken in state, we'll now have to modify github token in 2 places rather than have 1 source of truth. Why is githubtoken relevant to state?

Reply:
I agree. I didn't want this here. 

I originally had the functions strictly in github.js, but React is not able to detect state changes when you use localStorage. As a result, when you change the github token with the dialog box, the "add" button and "edit" button don't enable themselves unless you reload the page.